flink-streaming-java/src/main/java/org/apache/flink/streaming/api/graph/StreamGraph.java

```diff
@@ -85,6 +86,7 @@
 	private Set<Integer> sources;
 	private Set<Integer> sinks;
 	private Map<Integer, Tuple2<Integer, List<String>>> virtualSelectNodes;
+	private Map<Integer, Tuple2<Integer, OutputTag>> virtualOutputNodes;
```

Comment:
We might consider use  `addVirtualSideOutputNode`  and `virtualSideOutputNodes`. Unless we want to refactor move away from current assumption `<IN>operator<OUT>` to `<<tag1,IN1>...<tagX,INX> operator <<taga,OUTa>...<tagx,OUTX>`